🍒[cxx-interop] Include `Cxx` and `CxxStdlib` modules in no-stdlib builds

egorzhdan opened the pull request:
**Explanation**: This ensures that the `Cxx` and `CxxStdlib` binaries are compiled and installed correctly when building the stdlib separately from the compiler. These two modules are shipped with the toolchain, not with the SDK, so they need to be built along with the compiler. This also removes the unnecessary dependency on clang for these two modules.
**Scope**: Only affects clients which enable Swift/C++ interop.
**Risk**: Low: C++ interop is an opt-in feature that is enabled via a compiler flag. It is off by default.

rdar://107840627

egorzhdan committed:
[cxx-interop] Include `Cxx` and `CxxStdlib` modules in no-stdlib builds

These modules are shipped with the toolchain, while the stdlib might be built and shipped separately.

rdar://107780733
(cherry picked from commit d95215de5307a7e657293e1f2dc739ff3afe463b)

egorzhdan committed:
[cxx-interop] Do not add a dependency on clang to CxxStdlib

Cxx & CxxStdlib modules are Swift-only, they do not require invoking clang directly.

When building with `SWIFT_INCLUDE_TOOLS=NO`, Clang is not available as a CMake target (see `swift_common_standalone_build_config`).

rdar://107780733
(cherry picked from commit 14f32312bf03bd70274b2dc897dab96933db7b71)

zoecarver reviewed (approved):
I don't really know this code, but it looks great! Hopefully this fixes it 😅

compnerd committed:
build: add control of the build of `Cxx` module in -stdlib

This adds the ability to control the C++ interop module in the -stdlib
build separately from "extra toolchain content".  This is helpful for
Windows where we bootstrap the entire toolchain and do not have a stdlib
built when building tools.

(cherry picked from commit 435d4b4f8cfe97ab2a99882771aad5904f53d806)

egorzhdan commented:
@swift-ci please test

egorzhdan commented:
@swift-ci please test Linux
